src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -263,32 +275,44 @@ impl SourceScanInfo {
         Self::Incomplete(fetch_info)
     }
 
-    pub async fn complete(self) -> SchedulerResult<Self> {
+    pub async fn complete(self, batch_parallelism: usize) -> SchedulerResult<Self> {
```

Comment:
Is this `batch_parallelism` same as value in `new` function?

Reply:
This value is quite hard to pass to the `new` function, so I just pass where it needs.